Comorphisms/CASL2Prenex.hs

```diff
@@ -155,19 +156,13 @@ computePNF sen = case sen of
   Quantification q vdecls (computePNF qsen) nullRange
  Junction j sens _ -> let
    sens' = map computePNF sens
-   processQuants asen = case asen of
-     Quantification q vdecls sen' _ -> Quantification q vdecls (processQuants sen') nullRange
-     Junction j' jsens _ -> let
-       nonQuantifiedSen (Quantification _ _ _ _) = False
-       nonQuantifiedSen _ = True
-       (nonQuantJSens, remJSens) = span nonQuantifiedSen jsens
-       in if nonQuantJSens == jsens then asen
-           else case remJSens of 
-                  (Quantification q vdecls qsen _):remJSens' -> 
-                    Quantification q vdecls (processQuants $ Junction j' (nonQuantJSens ++ (qsen: remJSens')) nullRange) nullRange
-                  _ -> error "computePNF" -- should not be possible
-     _ -> asen
-  in processQuants $ Junction j sens' nullRange
+   (vdecls, sens'') = foldl (\(vs, ss) s -> case s of 
+                                           Quantification q vd' qs' _ -> ((q,vd'):vs, qs':ss)
+                                           _ -> (vs, s:ss)) 
+                    ([], []) sens'
+   qsen = Junction j (reverse sens'') nullRange
+   rsen = foldl (\s (q, vd) -> Quantification q vd s nullRange) qsen vdecls 
+  in rsen
```

Comment:
looks much better. However, it seems that looking into one level of quantification is not enough. Consider:
```
spec Mixed =
  sort s
  pred p,q:s
  op a,b:s
  . p(a) /\ ((forall x:s . p(x)) /\ (forall z:s . p(z)))  /\ (forall y:s . q(y))
end
```
One quantifier remains inside.